src/util/Options.js

```diff
@@ -175,20 +177,31 @@ class Options extends null {
   static cacheWithLimits(settings = {}) {
     const Collection = require('./Collection');
     const LimitedCollection = require('./LimitedCollection');
-    const SweptCollection = require('./SweptCollection');
 
     return manager => {
       const setting = settings[manager.name];
-      if (typeof setting === 'number' && setting !== Infinity) return new LimitedCollection(setting);
-      if (
-        /* eslint-disable-next-line eqeqeq */
-        (setting?.sweepInterval == null && setting?.sweepFilter == null) ||
+      /* eslint-disable-next-line eqeqeq */
+      if (setting == null) {
+        return new Collection();
+      }
+      if (typeof setting === 'number') {
+        if (setting === Infinity) {
+          return new Collection();
+        }
+        return new LimitedCollection({ maxSize: setting });
+      }
+      /* eslint-disable eqeqeq */
+      const noSweeping =
+        setting.sweepFilter == null ||
```

Comment:
`sweepFilter` is a function if i understand correctly?
```suggestion
        !setting.sweepFilter ||
```
```suggestion
        typeof setting.sweepFilter !== 'function' ||
```

Reply:
Better to have LimitedCollection constructor error for the other cases.